### PR TITLE
disable cloning for `Printable` class

### LINKED ISSUE #81: Set cloneable to `FALSE` for `Printable` class?
cf. https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/789#issuecomment-1055197820

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
closes #81

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
Not ready until https://github.com/Open-Systems-Pharmacology/OSPSuite.RUtils/pull/73 issue is resolved.

**Comment by PavelBal** [comment]
@abdullahhamadeh Do you use "clone" somewhere in the reporting engine? This would have an implication.

@IndrajeetPatil Then we have to check whether if we can enable cloning e.g. for `DataCombined`. Would be good,, but if not, I guess we have to live with it until the inheritance of the parameter is fixed.

**Comment by IndrajeetPatil** [comment]
> but if not, I guess we have to live with it until the inheritance of the parameter is fixed

Another option is to not inherit from `Printable`. 

We can have a custom print method for `DataComined` object because we aren't using much of `Printable`'s functionality anyway, so it's not worth making it a subclass of `Printable`.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# ospsuite.utils 1.2.0.9000
+
+MINOR CHANGES
+
+* Cloning method is now disabled for `Printable` R6 class.
```
[the remark this answers is not in the document]
Comment (IndrajeetPatil) [reply NEWS.md]: Done!

File: DESCRIPTION [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Type: Package
 Package: ospsuite.utils
 Title: Utility Functions for Open Systems Pharmacology R Packages
-Version: 1.2.0
+Version: 1.2.0.9000
```
[the remark this answers is not in the document]
Comment (msevestre) [reply DESCRIPTION:4]: or 1.2.0.9000 means what comes AFTER 1.2?

[the remark this answers is not in the document]
Comment (IndrajeetPatil) [reply DESCRIPTION:4]: https://r-pkgs.org/release.html?q=.9000#post-release

image

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+# ospsuite.utils 1.2.0.9000
+
+MAJOR CHANGES
+
+* Cloning method is now disabled for `Printable` R6 class. This entails that all
+  subclasses that inherit from this class will not be able to use cloning,
+  **even if** they explicitly set `cloneable = TRUE`.
```
Comment (PavelBal) [inline NEWS.md:7]: @IndrajeetPatil Did you actually **test** this? This is not the case!

Without overriding `cloneable`:

```
DataSet <- R6::R6Class(
  "DataSet",
  inherit = DotNetWrapper,
#  cloneable = FALSE,
  active = list(
```

```
> ds1 <- DataSet$new("myDs")
> ds1$clone()
DataSet: 
   Name: myDs 
   X dimension: Time 
   X unit: h 
   Y dimension: Concentration (mass) 
   Y unit: mg/l 
   Error type: 
   Error unit: 
   Molecular weight: 
   LLOQ: 
   Meta data: 
list()
```

With overriding:

```
DataSet <- R6::R6Class(
  "DataSet",
  inherit = DotNetWrapper,
  cloneable = FALSE,
  active = list(
```

```
> ds1 <- DataSet$new("myDs")
> ds1$clone()
Error: attempt to apply non-function
```


This documentation must be changed.

**Pull request merged**